### DISCUSSION TITLE
Need guidance for firmware porting and board bring up

### DISCUSSION THREAD

**Discussion by JustCurious8** [bitcraze/discussions#1037] [General]
Hi, we are trying to build and clone Crazyflie2.1 and bolt FCs from the open schematics, is there any proper documentation how to upload bootloaders and firmwares for nRF and STM32F05 microcontrollers for the first time, I'm not able to find any kind of board bring up or porting guide online, current bootloader and firmware flashing guide is not helping much

**Comment by ataffanel** [comment]
Hi,

No there is no such guide since this is not something supported by Bitcraze: Bitcraze is manufacturing and selling the Crazyflie, all software is open source and the schematic is published to allow customer to work in any way they want with the Crazyflie. The hardware files has not been published, so Crazyflie is not open-hardware. This is mostly due to the fact that we have had experience of bad quality Crazyflie 1 clones and that selling hardware is the Bitcraze's source of income so it is a bit scary to publish the hardware.

Now, we welcome experimentation with the schematics to make your own Crazyflie inspired drone, but boards bring-up is not something we can support on this forum. Similarly, hardware-specific support of non-bitcraze board will receive very limited support from Bitcraze itself: if you make your own hardware you need to support it.

I suggest you drop us an email at contact _at_ bitcraze.io to tell us a bit more what you are up to, we are open to collaborations that can be mutually beneficial.
